pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -471,6 +508,8 @@ func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *v1.Pod, podS
 			if pod.Spec.RestartPolicy != v1.RestartPolicyNever {
 				message := fmt.Sprintf("Failed to initialize pod. %q will be restarted.", container.Name)
 				glog.V(1).Info(message)
+
+				m.recorder.Eventf(containerRef, v1.EventTypeWarning, events.ContainerRestarting, message)
```

Comment:
See comment above, they're different audiences.  Kubelet is technically operating normally if it restarts dead pods.